MonoGame.Framework/Vector2.cs

```diff
@@ -756,9 +756,11 @@ public static void Negate(ref Vector2 value, out Vector2 result)
         /// </summary>
         public void Normalize()
         {
-            float val = 1.0f / (float)Math.Sqrt((X * X) + (Y * Y));
-            X *= val;
-            Y *= val;
+			var magnitude = (float)Math.Sqrt((X * X) + (Y * Y));
+			if(magnitude > 1E-05f)
```

Comment:
Is adhering to XNA really a top priority at this point? Wouldn't making the best possible framework for people to develop with today and in the future be a more important priority? At the very least it would be nice to have a method that "does the right thing" even if it has a slightly odd name. That coupled with a method summary explaining that NaN is a possible return from the method would at least save people some time who aren't coming from XNA and/or haven't run into trying to normalize 0,0 yet.